models/User.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * User Model
+ *
+ * Represents a registered user in the system.
+ * Handles identity, access, and relationships to projects, comments, etc.
+ */
+
+module.exports = (sequelize, Sequelize) => {
+  const { DataTypes } = Sequelize;
+
+  const User = sequelize.define(
+    "User",
+    {
+      id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
+        primaryKey: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
```

Comment:
Should we have name or firstname and lastname? For a silly app it probably doesn't matter. What do others think?

Reply:
I think we should add firstName and lastName, so we can re use these :)

Reply:
Added firstName,lastName (optional) and a virtual fullName.
```suggestion
      firstName: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      lastName: {
        type: DataTypes.STRING,
        allowNull: true,
      },
      fullName: {
        type: DataTypes.VIRTUAL,
        get() {
          return `${this.firstName} ${this.lastName || ""}`;
        },
      },
```